e2e-tests/playwright/specs/functional/system_console/permissions/team_access.spec.ts

```diff
@@ -0,0 +1,176 @@
+    await setupDefaultSystemManagerRole(
+        pw,
+        adminUser,
+        'permission_section_reporting',
+        'permission_section_reporting_team_statistics',
+        'Can edit',
+    );
+    await setupDefaultSystemManagerRole(
+        pw,
+        adminUser,
+        'permission_section_user_management',
+        'permission_section_user_management_teams',
+        'Can edit',
+    );
+
+    // Verify the system manager has access to the site statistics for all teams
+    await systemConsolePage.goto();
+
+    // Navigate to Team Statistics
+    await systemConsolePage.sidebar.goToItem('Team Statistics');
+
+    // Wait for page to fully load
+    await systemConsolePage.page.waitForLoadState('networkidle');
+
+    // Find the team filter dropdown
+    let teamFilterSelect = systemConsolePage.page.getByTestId('teamFilter');
+    await expect(teamFilterSelect).toBeVisible();
+
+    // Select the team by value
+    await teamFilterSelect.selectOption({value: team.id});
+
+    // Verify the text shows "Team Statistics for <team name>"
+    let teamStatsHeading = systemConsolePage.page.getByText(`Team Statistics for ${team.display_name}`, {exact: true});
+    await expect(teamStatsHeading).toBeVisible();
+
+    // Select the other team by value
+    await teamFilterSelect.selectOption({value: otherTeam.id});
+
+    // Verify the text shows "Team Statistics for <team name>"
+    const otherTeamStatsHeading = systemConsolePage.page.getByText(`Team Statistics for ${otherTeam.display_name}`, {
+        exact: true,
+    });
+    await expect(otherTeamStatsHeading).toBeVisible();
+
+    // Verify the user has API access to the otherTeam.
+    const fetchedOtherTeam = await systemManagerClient.getTeam(otherTeam.id);
+    expect(fetchedOtherTeam.id).toEqual(otherTeam.id);
+
+    // Configure the system manager without access to team user management
+    await setupDefaultSystemManagerRole(
+        pw,
+        adminUser,
+        'permission_section_user_management',
+        'permission_section_user_management_teams',
+        'No access',
+    );
+
+    // Verify the system manager only has access to the site statistics for the team they belong to
+    await systemConsolePage.goto();
+
+    // Navigate to Team Statistics
+    await systemConsolePage.sidebar.goToItem('Team Statistics');
+
+    // Find the team filter dropdown
+    teamFilterSelect = systemConsolePage.page.getByTestId('teamFilter');
+    await expect(teamFilterSelect).toBeVisible();
+
+    // Select the team by value
+    await teamFilterSelect.selectOption({value: team.id});
+
+    // Verify the text shows "Team Statistics for <team name>"
+    teamStatsHeading = systemConsolePage.page.getByText(`Team Statistics for ${team.display_name}`, {exact: true});
+    await expect(teamStatsHeading).toBeVisible();
+
+    // Verify the user has no API access to the otherTeam.
+    await expect(systemManagerClient.getTeam(otherTeam.id)).rejects.toThrow();
+});
```

Comment:
This is great. One potential suggestion (totally non-blocking) would be to optimize certain actions that can be performed in parallel through `promise.All`, such as creating the teams and roles. It may not seem much, but we were able to [speed up](https://github.com/mattermost/mattermost-plugin-calls/pull/995)  our Calls e2e tests by an interesting factor by doing this more often :)

Reply:
Ooh, nice trick! Will take a look :)